table/block_based/block_based_table_reader.cc

```diff
@@ -1352,12 +1355,21 @@ WithBlocklikeCheck<Status, TBlocklike> BlockBasedTable::GetDataBlockFromCache(
 
   // Lookup uncompressed cache first
   if (block_cache) {
-    BlockCreateContext create_ctx = rep_->create_context;
-    create_ctx.dict = dict;
     assert(!cache_key.empty());
-    auto cache_handle = block_cache.LookupFull(
-        cache_key, &create_ctx, GetCachePriority<TBlocklike>(), statistics,
-        rep_->ioptions.lowest_used_cache_tier);
+    typename BlockCacheInterface<TBlocklike>::TypedHandle* cache_handle;
+    if (decomp.get() != rep_->decompressor.get()) {
```

Comment:
Hmm why would this happen other than one of the two is nullptr? If so, why don't we always respect  rep_->decompressor?

Reply:
The comment on the next line mentions dict-aware decompressor. This is when they are different, if compression dictionary is in use. And the dict-aware decompressor lives on block cache (so that compression dictionaries can be swapped out), so it can't be part of the table reader. Added more commentary